src/japi.c

```diff
@@ -239,6 +253,11 @@ int japi_register_request(japi_context* ctx, const char *req_name, japi_req_hand
 		return -4;
 	}
 
+	if(ctx->init && strncmp(req_name, bad_req_name, strlen(bad_req_name)) == 0) {
+		fprintf(stderr, "ERROR: Request name is not allowed.\n");
+		return -5;
```

Comment:
The documentation in japi.h has to be updated.

Reply:
Thanks, done. Also I used a different error code for the two different cases